homeassistant/components/automation/__init__.py

```diff
@@ -160,21 +160,21 @@ def setup(hass, config):
     @asyncio.coroutine
     def trigger_service_handler(service_call):
         """Handle automation triggers."""
-        for entity in component.extract_from_service(service_call):
+        for entity in component.async_extract_from_service(service_call):
             hass.loop.create_task(entity.async_trigger(
                 service_call.data.get(ATTR_VARIABLES), True))
 
     @asyncio.coroutine
     def turn_onoff_service_handler(service_call):
         """Handle automation turn on/off service calls."""
         method = 'async_{}'.format(service_call.service)
-        for entity in component.extract_from_service(service_call):
+        for entity in component.async_extract_from_service(service_call):
             hass.loop.create_task(getattr(entity, method)())
 
     @asyncio.coroutine
```

Comment:
This handles calls to `async_turn_on` and `async_turn_off`.

 - Turn on needs to be a coroutine because it calls `prepare_setup_platform` inside an executor.
 - Turn off needs to be a coroutine because it calls `async_update_ha_state` which needs to be done at the end of the method because `async_remove` expects that.